Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAuditdPlugin.py

```diff
@@ -49,94 +68,299 @@ def Get_Marshall(Ensure):
         retd[k] = ld[k]
     return retval, retd
 
-def Set(Ensure):
-    # Add/Remove omsagent config for auoms
-    if Ensure == "Present":
-        text = ReadConf(RESOURCE_OMSAGENT_CONF)
-        WriteConf(OMSAGENT_CONF, text)
-    else:
-        RemoveConf()
+def IsUUID(uuidStr):
+    try:
+        uuidOut = uuid.UUID(uuidStr)
+    except:
+        return False
+    return str(uuidOut).encode('utf-8') == str(uuidStr).encode('utf-8')
+
+def IsValidWorkspaceId(WorkspaceId):
+    if not IsUUID(WorkspaceId):
+        return False
+    return os.path.isdir(os.path.join(AGENT_VAR_ROOT, WorkspaceId))
+
+def GetTmpDir(WorkspaceId):
+    return os.path.join(AGENT_VAR_ROOT, WorkspaceId, "tmp")
+
+def GetAgentConfDir(WorkspaceId):
+    return os.path.join(AGENT_ETC_ROOT, WorkspaceId, "conf/omsagent.d")
```

Comment:
Since you saved a lot of the other two-part "magic paths", why didn't you save this one?